nomad/api/base.py

```diff
@@ -19,23 +24,42 @@ def _endpointBuilder(self, *args):
             return "{v}/".format(v=self.version) + u
 
     def _urlBuilder(self, endpoint):
-        if self.verify or self.cert:
-            proto = 'https'
+        required_namespace=["job","jobs","allocation","allocations","deployment","deployments","acl"]
+        if not self.nspace:
+            url = "{uri}:{port}/{endpoint}".format(
+                uri=self.uri,
+                port=self.port,
+                endpoint=endpoint)
         else:
-            proto = 'http'
-
-        return "{proto}://{uri}:{port}/{endpoint}".format(
-            proto=proto,
-            uri=self.uri,
-            port=self.port,
-            endpoint=endpoint)
-
-    def get(self, endpoint, params=None):
+            if any(word in endpoint for word in required_namespace):
```

Comment:
This isn't totally safe; some endpoints might actually contain some of these keywords. Since so far there are no multi-level namespaces like `/status/peers` we can split ( kinda silly since we just constructed it ) and see what the first index is and see if it's within that list. If namespace support eventually comes to multi-level have to figure out a better method.

-